pkg/retry/retry.go

```diff
@@ -32,9 +32,15 @@ const (
 	DefaultMaxElapsedTime      = 1 * time.Minute
 )
 
-// Retry retries the given function DefaultRetryCount times while sleeping some time between unsuccessful attempts
+// Do retries the given function until DefaultMaxInterval time passes, while sleeping some time between unsuccessful attempts
 // if retryOnError returns true, continue retry, otherwise, return error
 func Do(operation func() error, retryOnError func(err error) bool) error {
+	return RetryWithConfiguredTime(operation, retryOnError, DefaultMaxElapsedTime)
+}
+
+// RetryWithConfiguredTime will retry with an exponential back off until "maxRetryTime" worth of sleeping has occurred.
+// this allows users of this function to customize how long they intend to retry a function before giving up on the operation.
+func RetryWithConfiguredTime(operation func() error, retryOnError func(err error) bool, maxRetryTime time.Duration) error {
```

Comment:
Just saw that the `backoff` is a pointer. How do you advise that we do the concurrent uploads with this? We will likely have to do them back to back unless we make two different exponentialBackoffs. Do you recommend this @caglar10ur ?